arch/xtensa/src/common/xtensa_hostfs.c

```diff
@@ -204,7 +204,7 @@ int host_unlink(const char *pathname)
   return -ENOSYS;
 }
 
-int host_mkdir(const char *pathname, mode_t mode)
+int host_mkdir(const char *pathname, int mode)
```

Comment:
why `int` here and not `nuttx_mode_t`?

Reply:
hostfs implementation on other platforms can reference nuttx header files directly without conflict, but sim can't do the same thing because there is huge conflict between host's libc and nuttx's. Please check  the code guarded by __SIM__ macro in include/nuttx/fs/hostfs.h.